src/panel-app/useImportMapOverrides.js

```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect } from "react";
+import { evalCmd } from "../inspected-window.helper.js";
+
+export default function useImportMapOverrides() {
+  const [importMapsEnabled, setImportMapEnabled] = useState(false);
+  const [overrides, setOverrides] = useState({});
+
+  async function checkImportMapOverrides() {
+    const hasImportMapsEnabled = await evalCmd(`(function() {
+      return !!window.importMapOverrides
+    })()`);
+    return hasImportMapsEnabled;
+  }
+
+  async function getImportMapOverrides() {
+    const { imports } = await evalCmd(`(function() {
+      return window.importMapOverrides.getOverrideMap()
+    })()`);
+    setOverrides(imports);
+  }
+
+  async function addOverride(currentMap, currentUrl) {
+    await evalCmd(`(function() {
+      return window.importMapOverrides.addOverride("${currentMap}", "${currentUrl}")
+    })()`);
+  }
+
+  async function removeOverride(currentMap) {
+    await evalCmd(`(function() {
+      return window.importMapOverrides.removeOverride("${currentMap}")
+    })()`);
+  }
+
+  async function batchSetOverrides() {
+    const overrideCalls = Object.entries(overrides).map(([map, url]) =>
+      !url ? removeOverride(map) : addOverride(map, url)
+    );
+    await Promise.all(overrideCalls);
+    await evalCmd(`window.location.reload()`);
+  }
+
+  // Get initial list of maps if they exist
+  useEffect(() => {
+    async function initImportMapsOverrides() {
+      const hasImportMapsEnabled = await checkImportMapOverrides();
```

Comment:
do we want to wrap with `try`? though we don't really do anything with errors at the moment anyway..... hmm. 

thoughts?

Reply:
Maybe I should add an ErrorBoundary in the app component instead? That way we can handle all errors, and maybe even add a link to GitHub issues.

Reply:
I like it.